internal/provider/resource_gitlab_personal_access_tokens.go

```diff
@@ -0,0 +1,242 @@
+		},
+		Schema: map[string]*schema.Schema{
+			"user_id": {
+				Description: "The id of the user.",
+				Type:        schema.TypeInt,
+				ForceNew:    true,
+				Required:    true,
+			},
+			"name": {
+				Description: "The name of the personal access token.",
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+			},
+			"scopes": {
+				Description: fmt.Sprintf("The scope for the personal access token. It determines the actions which can be performed when authenticating with this token. Valid values are: %s.", renderValueListForDocs(validPersonalAccessTokenScopes)),
+				Type:        schema.TypeSet,
+				Required:    true,
+				ForceNew:    true,
+				Elem: &schema.Schema{
+					Type:         schema.TypeString,
+					ValidateFunc: validation.StringInSlice(validPersonalAccessTokenScopes, false),
+				},
+			},
+			"active": {
+				Description: "True if the token is active.",
+				Type:        schema.TypeBool,
+				Computed:    true,
+			},
+			"revoked": {
+				Description: "True if the token is revoked.",
+				Type:        schema.TypeBool,
+				Computed:    true,
+			},
+			"created_at": {
+				Description: "Time the token has been created, RFC3339 format.",
+				Type:        schema.TypeString,
+				Computed:    true,
+			},
+			"expires_at": {
+				Description:      "The token expires at midnight UTC on that date. The date must be in the format YYYY-MM-DD. Default is never.",
+				Type:             schema.TypeString,
+				Optional:         true,
+				ForceNew:         true,
+				ValidateDiagFunc: isISO6801Date,
+			},
+			"token": {
+				Description: "The personal access token. This is only populated when creating a new personal access token. This attribute is not available for imported resources.",
+				Type:        schema.TypeString,
+				Computed:    true,
+				Sensitive:   true,
+			},
+		},
+	}
+})
+
+func resourceGitlabPersonalAccessTokenCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(*gitlab.Client)
+
+	userId := d.Get("user_id").(int)
+
+	currentUserAdmin, err := isCurrentUserAdmin(client)
+	if err != nil {
+		return diag.Errorf("[ERROR] cannot query the user API for current user: %v", err)
+	}
+
+	if !currentUserAdmin {
+		return diag.Errorf("[ERROR] Current user needs to be admin when creating a personal access token: %v", err)
+	}
+
+	options := &gitlab.CreatePersonalAccessTokenOptions{
+		Name:   gitlab.String(d.Get("name").(string)),
+		Scopes: stringSetToStringSlice(d.Get("scopes").(*schema.Set)),
+	}
+
+	log.Printf("[DEBUG] create gitlab PersonalAccessToken %s (scopes: %s) for user ID %d", *options.Name, options.Scopes, userId)
+
+	if v, ok := d.GetOk("expires_at"); ok {
+		parsedExpiresAt, err := time.Parse("2006-01-02", v.(string))
+		if err != nil {
+			return diag.Errorf("Invalid expires_at date: %v", err)
+		}
+
+		parsedExpiresAtISOTime := gitlab.ISOTime(parsedExpiresAt)
+		options.ExpiresAt = &parsedExpiresAtISOTime
```

Comment:
we have a helper function for that:

```suggestion
		```suggestion
		parsedExpiresAt, err := parseISO8601Date(v.(string))
		if err != nil {
			return diag.Errorf("failed to parse expires_at '%s' as ISO8601 formatted date: %v", v.(string), err)
		}
		
		options.ExpiresAt = parsedExpiresAt
```
```

Reply:
Implemented